### PR TITLE
Update build and test with mxnet 1.7.0

### PR DESCRIPTION
**Pull request by sxjscience**
Test for MXNet 1.7.0 nightly to see if there are any issue.

### CODE REVIEW

File: docs/build.yml [hunk 1]
```diff
@@ -3,8 +3,9 @@ dependencies:
 - python=3.7
 - graphviz
 - pip:
+  - --find-links https://dist.mxnet.io/python
   - jupyter-sphinx>=0.2.2
-  - mxnet-cu100==1.6.0b20191013
+  - mxnet-cu100==1.7.0b20200629
```
Comment (leezu) [inline docs/build.yml:8]: I believe this 1.7.0b actually is a build of the 1.x branch.

**Comment by sxjscience** [comment]
@ciyongch We are going to test AutoGluon with MXNet 1.7. Currently, I'm using the nightly builds of 1.x but will update that to 1.7.rc0 once that's ready.

**Comment by ciyongch** [comment]
Thanks for tagging me on this @sxjscience , I'm not sure whether there'll be nightly image for v1.7.x or later v1.7.rc0. Anyway, I will keep you posted when 1.7.rc0 is announced.

**Comment by ciyongch** [comment]
Hi @sxjscience , MXNet 1.7.0.rc0 is cut off, would you mind to have a try with the latest code base? Thanks.

**Comment by sxjscience** [comment]
@ciyongch Would you know how should I try it in CI?

**Comment by ciyongch** [comment]
push an [empty] commit?

**Comment by sxjscience** [comment]
I actually mean if there is any pip package because the previous one seems to be the 1.x nightly build. I’m currently rebuilding the static pip package.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by ciyongch** [comment]
Sorry for misunderstood your requirement. Probably there's no 1.7.0.rc0 (or 1.7.x branch) pip wheel yet, ping @szha @leezu to double confirm.

**Comment by sxjscience** [comment]
@ciyongch Basically the CI fails by throwing `weakref` not pickleble error. It might be due to the fact the gluon block is not pickleble. But I need to investigate the root cause.

**Comment by szha** [comment]
Job PR-546-8 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-546/8/index.html

**Comment by sxjscience** [comment]
@ciyongch The CI of AutoGluon has passed after the PR in MXNet is reverted

**Comment by sxjscience** [comment]
@Jerryzcn @zhreshold @jwmueller @Innixma 

We are able to pass the test with 1.7.0. Currently, I'm using the wheels built from the 1.7.x branch in MXNet:

The wheel packages used:
- CPU
   - https://autogluon-kdd-mxnet-wheels.s3-accelerate.amazonaws.com/mxnet-1.7.0b20200713-py2.py3-none-manylinux2014_x86_64.whl
- GPU + CUDA100
   -  https://autogluon-kdd-mxnet-wheels.s3-accelerate.amazonaws.com/mxnet_cu100-1.7.0b20200713-py2.py3-none-manylinux2014_x86_64.whl

**Comment by Innixma** [comment]
Great! Is it ok if we wait until 1.7.0 is officially released before merging this PR?

**Comment by sxjscience** [comment]
@Innixma Yes, it's fine to wait until the official 1.7.0 so I will leave it open for now.

**Comment by zhreshold** [comment]
@sxjscience Good job, thanks for pushing this

**Comment by ciyongch** [comment]
It's great to root cause the issue, thanks @sxjscience !
@leezu would you mind to revert the https://github.com/apache/incubator-mxnet/pull/18358 for 1.7.0 release? Thanks.

**Comment by szha** [comment]
Job PR-546-10 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-546/10/index.html

**Comment by szha** [comment]
Job PR-546-11 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-546/11/index.html

**Comment by szha** [comment]
Job PR-546-12 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-546/12/index.html

File: docs/build.yml [hunk 2]
```diff
@@ -3,8 +3,10 @@ dependencies:
 - python=3.7
 - graphviz
 - pip:
+  - --find-links https://sxjscience.github.io/KDD2020/
   - jupyter-sphinx>=0.2.2
-  - mxnet-cu100==1.6.0b20191013
+  - gluonnlp==0.8.1
+  - mxnet_cu101>=1.7.0b20200713, <2.0.0
```
Comment (Innixma) [inline docs/build.yml:9]: Do we think it would be wise to limit to <1.8.0 ?

File: docs/install-include.rst [hunk 3]
```diff
@@ -58,16 +58,17 @@ Select your preferences below and run the corresponding install commands:
 
               .. code-block:: bash
 
-                 pip install --upgrade mxnet
+                 pip install -U --pre "mxnet>=1.7.0b20200713, <2.0.0" -f https://sxjscience.github.io/KDD2020/
```
Comment (Innixma) [inline docs/install-include.rst:61]: Is there a more official way to install this than using https://sxjscience.github.io/KDD2020/?

Comment (sxjscience) [reply docs/install-include.rst:61]: Currently the 1.7.0 vote hasn't been passed yet so we have to use this form temporarily.

File: docs/build.yml (hunk 2 above)
Comment (sxjscience) [reply docs/build.yml:9]: I think versions less than 2.0.0 should be backward compatible so it should be fine.

Comment (szha) [reply docs/build.yml:9]: mxnet follows semantic versioning, so there shouldn't be any breakage in 1.x unless there's an issue in mxnet

**Comment by szha** [comment]
Job PR-546-13 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-546/13/index.html

**Pull request merged**